src/session/session_api.c

```diff
@@ -1668,14 +1669,25 @@ __wt_session_range_truncate(
     trunc_info->orig_stop_key = orig_stop_key;
     trunc_info->uri = actual_uri;
 
+    if (start != NULL && strcmp(start->uri, WT_HS_URI) == 0)
+        start_file_cursor = ((WT_CURSOR_HS *)start)->file_cursor;
+    else
+        start_file_cursor = start;
+
+    if (stop != NULL && strcmp(stop->uri, WT_HS_URI) == 0)
+        stop_file_cursor = ((WT_CURSOR_HS *)stop)->file_cursor;
+    else
+        stop_file_cursor = stop;
+
     /*
      * Don't use bounded cursors for FLCS as it isn't supported, additionally skip using them for
      * complex types such as column groups and indexes. We can't check support for those complex
      * types at this abstraction level.
      */
-    if (CUR2BT(start) == NULL || CUR2BT(start)->type == BTREE_COL_FIX)
+    if (CUR2BT(start_file_cursor) == NULL || CUR2BT(start_file_cursor)->type == BTREE_COL_FIX)
```

Comment:
It is a little confusing to introduce two new cursor variables just for this check. I'd say a better solution would be to incorporate checking for HS directly into these two `if` statements, such as something like this:

```
if (strcmp(start->uri, WT_HS_URI) == 0 || CUR2BT(start) == NULL || CUR2BT(start)->type == BTREE_COL_FIX)
```

(I don't think HS cursors support bounds—pls correct me if I'm wrong.) Adding a check here would make the condition more readable, and it would eliminate the need for the two new variables.

However, I still don't like this solution—a much better solution would check that `start` is a {{WT_CURSOR_BTREE}}. This function is called directly from `__session_truncate` (which is the public API `WT_SESSION::truncate`), so users can pass any kinds of cursors they want as long as they support `compare`. Ideally, this would become something like this:

```
if (!WT_CURSOR_IS_BTREE(start) || CUR2BT(start) == NULL || CUR2BT(start)->type == BTREE_COL_FIX)
```

...except that I don't know how to reliably check for the cursor type. Which I guess brings me to my other comments, where I wrote that I wish that `WT_CURSOR` would have a type field.

Alternatively, I wonder if there is a more generic way to check whether bounds are supported. I wish we could do something like `start->bound != __wt_cursor_config_notsup`, but that wouldn't work for complex types (column groups, tables with indexes). I wonder if it would make sense to have flags or some other ways to check that bounds are supported?